5-AccessControl/2-call-api-groups/SPA/src/components/RouteGuard.jsx

```diff
@@ -1,63 +1,81 @@
-import React, { useState, useEffect } from "react";
-import { Route, Redirect } from "react-router-dom";
-import { useMsal } from "@azure/msal-react";
+import { useState, useEffect } from 'react';
+import { useMsal, MsalAuthenticationTemplate } from '@azure/msal-react';
+import { InteractionType } from '@azure/msal-browser';
+import { loginRequest } from '../authConfig';
+import { useLocation, Navigate } from 'react-router-dom';
 
 export const RouteGuard = ({ Component, ...props }) => {
+    const location = useLocation();
 
     const { instance } = useMsal();
     const [isAuthorized, setIsAuthorized] = useState(false);
     const [isOveraged, setIsOveraged] = useState(false);
 
-    const onLoad = async () => {
-        if (props.location.state) {
-            let intersection = props.groups
-                .filter(group => props.location.state.groupsData.includes(group));
+    const authRequest = {
+        ...loginRequest,
+    };
 
+    const onLoad = async () => {
+        if (location.state && location.state.groupsData) {
+            let intersection = props.groups.filter((group) => location.state.groupsData.includes(group));
             if (intersection.length > 0) {
                 setIsAuthorized(true);
             }
         } else {
             const currentAccount = instance.getActiveAccount();
-
             if (currentAccount && currentAccount.idTokenClaims['groups']) {
-                let intersection = props.groups
-                    .filter(group => currentAccount.idTokenClaims['groups'].includes(group));
+                let intersection = props.groups.filter((group) =>
+                    currentAccount.idTokenClaims['groups'].includes(group)
+                );
 
                 if (intersection.length > 0) {
                     setIsAuthorized(true);
                 }
-            } else if (currentAccount && (currentAccount.idTokenClaims['_claim_names'] || currentAccount.idTokenClaims['_claim_sources'])) {
-                window.alert('You have too many group memberships. The application will now query Microsoft Graph to get the full list of groups that you are a member of.');
+            } else if (
+                currentAccount &&
+                (currentAccount.idTokenClaims['_claim_names'] ||
+                    (currentAccount.idTokenClaims['_claim_sources'] && !isOveraged))
```

Comment:
```suggestion
                (currentAccount.idTokenClaims['_claim_names'] ||
                    (currentAccount.idTokenClaims['_claim_sources'] && !isOveraged))
```

See the guidelines in README for what needs the be checked in this if clause:

> Check for the claim _claim_names with one of the values being **groups**. This indicates overage.